app/models/post.rb

```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :user, foreign_key: 'user_id'
+  has_many :comments
+  has_many :likes
+
+  def update_counter(user_id)
+    user = User.find_by(id: user_id)
+    user.posts_counter = Post.where(user_id: user_id).count
+    user.save
+  end
```

Comment:
Great job @Progress2002 🤝

- [ ] [Optional] I would suggest you use the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_counter` method every time a new post has been saved to the database, then make the `update_counter` method private.
- [ ] Kindly repeat this across all `#update_counter` methods in the `like` and `comment` models.

Reply:
Hello @1GORDON , thanks for taking the time to review and approve my project. In regards to the optional suggestion, it's quite a cool feature to implement thanks for the tips but I will like to bring to your notice that based on the project requirements for this mileston, we are not expected to implement this feature just yet.